courses-app/src/App.tsx

```diff
@@ -66,91 +68,63 @@ function App() {
 			navigate('/courses');
 	}, [token, location, navigate]);
 
-	useEffect(() => {
-		if (localStorage.getItem('token')) {
-			dispatch(
-				loginUser({
-					successful: true,
-					result: localStorage.getItem('token') as string,
-					user: {
-						name: localStorage.getItem('userName') as string,
-						email: localStorage.getItem('email') as string,
-					},
-				})
-			);
-			dispatch(getCourses(mockedCoursesList));
-			dispatch(getAuthors(mockedAuthorsList));
+	const handleLogin = () => {
+		if (token) {
+			dispatch(getCoursesRequest());
+			dispatch(getAuthorsRequest());
 		}
-	}, [dispatch]);
-
-	const handleLogin = (response: TokenResponse) => {
-		const coursesList = async () => {
-			try {
-				const result = await CoursesAPI.getCourses();
-				dispatch(getCourses(result.result));
-				return result;
-			} catch (error) {
-				dispatch(getCourses(mockedCoursesList));
-				return error;
-			}
-		};
-		const authorsList = async () => {
-			try {
-				const result = await AuthorsAPI.getAuthors();
-				dispatch(getAuthors(result.result));
-				return result;
-			} catch (error) {
-				dispatch(getAuthors(mockedAuthorsList));
-				return error;
-			}
-		};
-		coursesList();
-		authorsList();
-		dispatch(loginUser(response));
 	};
 
 	const handleLogout = () => {
-		localStorage.removeItem('token');
-		localStorage.removeItem('userName');
-		localStorage.removeItem('email');
-		dispatch(logoutUser());
-	};
-
-	const creationCoursesHandler = (
-		course: CourseType,
-		courseAuthors: AuthorType[]
-	) => {
-		dispatch(addCourse(course));
-		dispatch(
-			addAuthor([
-				...courseAuthors.filter(
-					(atr) => !allAuthors.some((haveAuthor) => haveAuthor.id === atr.id)
-				),
-			])
-		);
-		navigate('/courses');
+		dispatch(logOutRequest(token));
 	};
 
 	return (
 		<SharedContext.Provider
 			value={{
 				userName,
+				role,
 				handleLogout,
 				token,
 				isLoading,
+				isAuth,
 				setIsLoading,
 				courses,
 				allAuthors,
-				creationCoursesHandler,
 				handleLogin,
+				userError,
+				userLoading,
+				coursesError,
+				coursesLoading,
 			}}
```

Comment:
Just noticed you are still using the context. all of this can be stored in redux

Reply:
Remove context